### PR TITLE
Allow plain pg foreign tables without a table_name option

### LINKED ISSUE #6642: CREATE FOREIGN TABLE fails if the option "table_name" is not provided.
Env: tools/citus_dev/citus_dev make testCluster

psql -p 9700

> 
> CREATE EXTENSION **postgres_fdw;**
> 
> CREATE SERVER foreign_server
>           FOREIGN DATA WRAPPER postgres_fdw
>           OPTIONS (host 'localhost', port '9700', dbname 'regression');
> 
> CREATE USER MAPPING FOR CURRENT_USER
>         SERVER foreign_server
>         OPTIONS (user 'emel');
> 
> CREATE FOREIGN TABLE foreign_table (
>         id integer NOT NULL,
>         data text,
>         a bigserial
> )
>         SERVER foreign_server
>         OPTIONS (schema_name 'foreign_tables_schema_mx');

**ERROR:  table_name option must be provided when using postgres_fdw with Citus
HINT:  Provide the option "table_name" with value target table's name**

Looking at the citus regression tests (foreign_tables_mx.sql) , this seems to be expected when
  SET citus.use_citus_managed_tables TO ON;

However, the above commands fail even if there is no citus configuration and  citus is only loaded.

Citus utility hook seems to interfere somehow.

This bug can be confirmed by running regression tests of the extension
postgresql-14.6/contrib/postgres_fdw
postgresql-15.1/contrib/postgres_fdw

by simply loading Citus during their regression runs.

### PR DESCRIPTION
**Pull request by gokhangulbiz**
Fixes #6642

### CODE REVIEW

File: src/test/regress/sql/foreign_tables_mx.sql [hunk 1]
```diff
@@ -219,14 +227,21 @@ SELECT * FROM ref_tbl d JOIN foreign_table_local f ON d.a=f.id ORDER BY f.id;
...
         data text
 )
         SERVER foreign_server_local
         OPTIONS (schema_name 'foreign_tables_schema_mx');
 
+-- should error out because doesn't have a table_name field
+SELECT citus_add_local_table_to_metadata('foreign_table_local_fails');
+
+-- should work since it has a table_name
+ALTER FOREIGN TABLE foreign_table_local_fails OPTIONS (table_name 'foreign_table_local_fails');
```
Comment (aykut-bozkurt) [inline src/test/regress/sql/foreign_tables_mx.sql:241]: NIT: Can we change table_name to something different than 'foreign_table_local_fails' which is also foreign table's name. There is no query for the table but that created a self loop and consumed all connections on my local.

File: src/backend/distributed/commands/citus_add_local_table_to_metadata.c [hunk 2]
```diff
@@ -509,6 +520,48 @@ ErrorIfUnsupportedCreateCitusLocalTable(Relation relation)
...
 
+/*
+ * EnsureIfFdwHasTableName errors out with a hint if the foreign table is using postgres_fdw and
+ * the option table_name is not provided.
+ */
+static void
+EnsureIfFdwHasTableName(Oid relationId)
+{
+	char relationKind = get_rel_relkind(relationId);
+	if (relationKind == RELKIND_FOREIGN_TABLE)
```
Comment (onderkalaci) [inline src/backend/distributed/commands/citus_add_local_table_to_metadata.c:531]: should/can we do this only for postgres_fdw? I believe the code used to only check for postgres_fdw, but I cannot see we filter that here (maybe I'm missing something)?

**Comment by aykut-bozkurt** [comment]
We add foreign table to metadata only if its `table_name` option is set. Then, when we drop `table_name` option, should we also drop foreign table from metadata. Otherwise, we got an error like below as table name defaults to foreign table's name. 

```sql
ERROR:  relation "public.foreign_table_local_fails2_102013" does not exist
CONTEXT:  remote SQL command: SELECT id, data FROM public.foreign_table_local_fails2_102013
```

**Comment by onderkalaci** [comment]
> ALTER FOREIGN TABLE foreign_table options  (drop table_name);  -- still exists in metadata

Can you open another issue for that? I guess we should disallow drop table_name option for postgres_fdw.

> I think that occurs before the PR. I donot know if we should fix it here. @marcocitus

We should fix that, but in a separate PR? I hope/expect the fix to be few lines of code?

File: src/backend/distributed/commands/citus_add_local_table_to_metadata.c (hunk 2 above)
Comment (gokhangulbiz) [reply src/backend/distributed/commands/citus_add_local_table_to_metadata.c:531]: I think that's a valid case for any other extension that supports foreign tables. So I wouldn't say we should add a check for only postgres_fdw.

**Comment by aykut-bozkurt** [comment]
> > ALTER FOREIGN TABLE foreign_table options  (drop table_name);  -- still exists in metadata
> 
> Can you open another issue for that? I guess we should disallow drop table_name option for postgres_fdw.
> 
> > I think that occurs before the PR. I donot know if we should fix it here. @marcocitus
> 
> We should fix that, but in a separate PR? I hope/expect the fix to be few lines of code?

https://github.com/citusdata/citus/issues/6663

File: src/backend/distributed/commands/citus_add_local_table_to_metadata.c (hunk 2 above)
Comment (gokhangulbiz) [reply src/backend/distributed/commands/citus_add_local_table_to_metadata.c:551]: I was wrong :) After we discussed this with Onder in private we agreed to make this check only for postgres_fdw.

**Review by aykut-bozkurt — APPROVED** [review]

**Pull request merged**